src/rgw/services/svc_notify.cc

```diff
@@ -377,107 +381,54 @@ int RGWSI_Notify::distribute(const DoutPrefixProvider *dpp, const string& key, b
     RGWSI_RADOS::Obj notify_obj = pick_control_obj(key);
 
     ldpp_dout(dpp, 10) << "distributing notification oid=" << notify_obj.get_ref().obj
-        << " bl.length()=" << bl.length() << dendl;
-    return robust_notify(dpp, notify_obj, bl, y);
+		       << " cni=" << cni << dendl;
+    return robust_notify(dpp, notify_obj, cni, y);
   }
   return 0;
 }
 
-int RGWSI_Notify::robust_notify(const DoutPrefixProvider *dpp, 
-                                RGWSI_RADOS::Obj& notify_obj, bufferlist& bl,
+int RGWSI_Notify::robust_notify(const DoutPrefixProvider *dpp,
+                                RGWSI_RADOS::Obj& notify_obj,
+				const RGWCacheNotifyInfo& cni,
                                 optional_yield y)
 {
   // The reply of every machine that acks goes in here.
   boost::container::flat_set<std::pair<uint64_t, uint64_t>> acks;
-  bufferlist rbl;
+  bufferlist bl;
+  encode(cni, bl);
 
   // First, try to send, without being fancy about it.
-  auto r = notify_obj.notify(dpp, bl, 0, &rbl, y);
+  auto r = notify_obj.notify(dpp, bl, 0, nullptr, y);
 
-  // If that doesn't work, get serious.
   if (r < 0) {
-    ldpp_dout(dpp, 1) << "robust_notify: If at first you don't succeed: "
-		  << cpp_strerror(-r) << dendl;
-
-
-    auto p = rbl.cbegin();
-    // Gather up the replies to the first attempt.
-    try {
-      uint32_t num_acks;
-      decode(num_acks, p);
-      // Doing this ourselves since we don't care about the payload;
-      for (auto i = 0u; i < num_acks; ++i) {
-	std::pair<uint64_t, uint64_t> id;
-	decode(id, p);
-	acks.insert(id);
-	ldpp_dout(dpp, 20) << "robust_notify: acked by " << id << dendl;
-	uint32_t blen;
-	decode(blen, p);
-	p += blen;
-      }
-    } catch (const buffer::error& e) {
-      ldpp_dout(dpp, 0) << "robust_notify: notify response parse failed: "
-		    << e.what() << dendl;
-      acks.clear(); // Throw away junk on failed parse.
-    }
-
-
-    // Every machine that fails to reply and hasn't acked a previous
-    // attempt goes in here.
-    boost::container::flat_set<std::pair<uint64_t, uint64_t>> timeouts;
+    ldpp_dout(dpp, 1) << __PRETTY_FUNCTION__ << ":" << __LINE__
+		      << " Notify failed: "
+		      << cpp_strerror(-r) << dendl;
+    ClibBackTrace bt(1);
+    ldpp_dout(dpp, 1) << __PRETTY_FUNCTION__ << ":" << __LINE__
+		      << " Backtrace of the operation where this happened: "
+		      << bt << dendl;
```

Comment:
i don't really see the utility of logging a backtrace when we get errors from notify. that seems to imply that the issue is somewhere in the local rgw's call stack, but notify errors are generally due to issues elsewhere in the cluster - like network issues, or other radosgws or osds being slow

the dpp already includes information about the associated request. could we just add the object name to the `Notify failed:` message? that should be enough to correlate the error with other osd/radosgw logs

Reply:
That sounds reasonable post-DPP.

Do we want the callstack pre-DPP or is it not worth the trouble?

Reply:
(Pre-DPP as in backports.)

Reply:
up to you; would that have made it easier for you to track this one down?

Reply:
Probably would have been a clue, so I'll stick that in on backport.